File: src/main/java/com/pubnub/api/models/consumer/pubsub/objects/PNSpaceResult.java
package com.pubnub.api.models.consumer.pubsub.objects;


import com.pubnub.api.models.consumer.objects_api.space.PNSpace;
import com.pubnub.api.models.consumer.pubsub.BasePubSubResult;
import lombok.Builder;
import lombok.ToString;

@ToString(callSuper = true)
public class PNSpaceResult extends ObjectResult<PNSpace> {

    @Builder(builderMethodName = "spaceBuilder")
    private PNSpaceResult(BasePubSubResult result, String event, PNSpace space) {
        super(result, event, space);
    }

    public PNSpace getSpace() {
        return data;
    }
}

File: src/main/kotlin/com/pubnub/api/models/consumer/pubsub/objects/PNSpaceResult.kt
package com.pubnub.api.models.consumer.pubsub.objects

import com.google.gson.JsonElement
import com.pubnub.api.models.consumer.pubsub.BasePubSubResult

class PNSpaceResult(
    override val result: BasePubSubResult,
    override val event: String,
    override val data: JsonElement
) : ObjectResult<JsonElement>(
    result,
    event,
    data
)